src/Compilers/Core/Portable/PEWriter/SigningUtilities.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Security.Cryptography;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Emit;
+
+namespace Microsoft.Cci
+{
+    internal static class SigningUtilities
+    {
+        internal static byte[] CalculateRsaSignature(IEnumerable<Blob> content, RSAParameters privateKey)
+        {
+            var hash = CalculateSha1(content);
+
+            using (var rsa = RSA.Create())
+            {
+                rsa.ImportParameters(privateKey);
+                var signature = rsa.SignHash(hash, HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);
+                Array.Reverse(signature);
+                return signature;
+            }
+        }
+
+        private static byte[] CalculateSha1(IEnumerable<Blob> content)
+        {
+            using (var hash = IncrementalHash.CreateHash(HashAlgorithmName.SHA1))
+            {
+                var stream = new MemoryStream();
```

Comment:
Is `stream` needed?

Reply:
Huh yeah, it's not used in [the original](http://source.dot.net/#System.Reflection.Metadata.Tests/TestUtilities/SigningUtilities.cs,d073044413519e36,references) either.

Done.